gensim/models/keyedvectors.py

```diff
@@ -557,6 +563,119 @@ def most_similar(self, positive=None, negative=None, topn=10, restrict_vocab=Non
         # ignore (don't return) words from the input
         result = [(self.index2word[sim], float(dists[sim])) for sim in best if sim not in all_words]
         return result[:topn]
+        
+    def evaluate_word_analogies_setBased(self, analogies, restrict_vocab=300000, case_insensitive=True, dummy4unknown=False, topk=1):
+        """Compute performance of the model on an analogy test set.
+        This is modern variant of :meth:`~gensim.models.keyedvectors.WordEmbeddingsKeyedVectors.accuracy`, see
+        `discussion on GitHub #1935 <https://github.com/RaRe-Technologies/gensim/pull/1935>`_.
+        The accuracy is reported (printed to log and returned as a score) for each section separately,
```

Comment:
Instead of copy-paster docstring form another method, need to 
- make a reference to the original method
- describe, why the current method different, what's new here, why is that useful

Reply:
Fixed